core/domain/suggestion_registry.py

```diff
@@ -421,24 +424,26 @@ def __init__(
             exp_domain.ExplorationChange(change)
         )
         self.score_category = score_category
-        # In BaseSuggestion, language_code is defined with only string type
-        # but here language_code is of Optional[str] type because language_code
-        # can accept None values as well. So, due to this conflict in types
-        # MyPy throws an `Incompatible types in assignment` error. Thus to
-        # avoid the error, we used ignore here.
+        # Here we use MyPy ignore because in BaseSuggestion, language_code
+        # is defined with only string type but here language_code is of
+        # Optional[str] type because language_code can accept None values as
+        # well. So, due to this conflict in types MyPy throws an `Incompatible
+        # types in assignment` error. Thus to avoid the error, we used ignore.
         self.language_code = language_code  # type: ignore[assignment]
-        # In BaseSuggestion, last_updated is defined with only datetime type
-        # but here last_updated is of Optional[datetime] type because
-        # last_updated can accept None values as well. So, due to this conflict
-        # in types MyPy throws an `Incompatible types in assignment` error.
-        # Thus to avoid the error, we used ignore here.
+        # Here we use MyPy ignore because in BaseSuggestion, last_updated
+        # is defined with only datetime type but here last_updated is of
+        # Optional[datetime] type because last_updated can accept None
+        # values as well. So, due to this conflict in types MyPy throws an
+        # `Incompatible types in assignment` error. Thus to avoid the error,
+        # we used ignore here.
         self.last_updated = last_updated  # type: ignore[assignment]
         self.edited_by_reviewer = edited_by_reviewer
-        # In BaseSuggestion, image_context is defined as string type attribute
-        # but currently, we don't allow adding images in the "edit state
-        # content" suggestion, so the image_context is None here and due to
-        # None MyPy throws an `Incompatible types in assignment` error.
-        # Thus to avoid the error, we used ignore here.
+        # Here we use MyPy ignore because in BaseSuggestion, image_context
```

Comment:
Then shouldn't we fix it in BaseSuggestion (with explanation) and give it more specific types in the relevant subclasses? (Ditto elsewhere, re giving more specific types in the subclasses.)

Also the justification seems wrong, it's possible to make content suggestions with images. So, may need more detail on when this is equal to None exactly.

Reply:
Image_context can only be None in ( here none is not added by me, it is added already before my changes in the codebase  ) https://github.com/oppia/oppia/blob/68bb129c7fe773b989670a59162534b8893789bb/core/domain/suggestion_registry.py#L389

While in all other sub-classes It is not None -
https://github.com/oppia/oppia/blob/68bb129c7fe773b989670a59162534b8893789bb/core/domain/suggestion_registry.py#L675
https://github.com/oppia/oppia/blob/68bb129c7fe773b989670a59162534b8893789bb/core/domain/suggestion_registry.py#L900

So, if we use Optional[str] in BaseSuggestion then we have to use more if else exception clauses unnessesarly to narrow down the type in all other suggestions.